akd/src/tests.rs

```diff
@@ -130,6 +131,39 @@ fn setup_mocked_db(db: &mut MockLocalDatabase, test_db: &AsyncInMemoryDatabase)
         });
 }
 
+// A test to ensure that any database error at the time a Directory is created
+// does not automatically attempt to create a new aZKS. Only aZKS not found errors
+// should assume that a successful read happened and no aZKS exists.
+test_config!(test_directory_azks_bootstrapping);
+async fn test_directory_azks_bootstrapping<TC: Configuration>() -> Result<(), AkdError> {
+    let vrf = HardCodedAkdVRF {};
+
+    // Verify that a Storage error results in an error when attempting to create the Directory
+    let mut mock_db = MockLocalDatabase {
+        ..Default::default()
+    };
+    mock_db
+        .expect_get::<Azks>()
+        .returning(|_| Err(StorageError::Connection("Fire!".to_string())));
+    let storage = StorageManager::new_no_cache(mock_db);
+
+    let maybe_akd = Directory::<TC, _, _>::new(storage, vrf.clone()).await;
+    assert!(maybe_akd.is_err());
+
```

Comment:
Can we also check here that `storage.get` doesn't return any data for `AZKS` and `root` `StorageKey`s to ensure they haven't been inserted into the database?

Reply:
The mock is configured to return an `Err` here, so I don't really think that's possible in this case.

Reply:
Can we confirm mock database's set function is not called perhaps? (If not I'm fine as is.)

Reply:
@eozturk1 good call! Added.